pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -823,6 +827,59 @@ public String uploadLLCSegment(
     }
   }
 
+  @POST
+  @Path("/tables/{realtimeTableName}/consumingSegmentsFreshnessInfo")
+  @Authorize(targetType = TargetType.TABLE, paramName = "realtimeTableName",
+      action = Actions.Table.GET_SEGMENTS_FRESHNESS)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Returns ingestion timestamp ms for given segments",
+      notes = "Any segments that are not consuming will be excluded")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 404, message = "Table not found"),
+      @ApiResponse(code = 500, message = "Internal server error")
+  })
+  public Map<String, Long> getSegmentsFreshnessInfo(
```

Comment:
I think we want to get the table level freshness value (minimum of freshness value across all streaming partitions) and return it as one single long value. Do you see the need of return per segment freshness? Ideally it should be directly pulled from `RealtimeTableDataManager`

Reply:
you're right that the minimum freshness computation can be done here to avoid having to push the map of the segments as part of the response

this does go against the idea of having partition level freshness info for debugging purposes. What do you suggest?

Reply:
I guess we can provide 2 APIs (or combining 2 functionalities into one API), one returning the per partition info, one return the table level info. The API should only take the table name as input